0.20

balloob opened the pull request:
Tons of new supported things in 0.20.
- Restarting now supported without forking (@jaharkes)
- Add purge days option to recorder component (@justyns, @infamy)
- Generic load_platform mechanism for entity components (@kellerza)
- Template: new `relative_time` function to render ie. `7 seconds ago` (@robbiet480)
- Sensor: supervisord now supported (@happyleavesaoc)
- Docker: we should no longer get SSL errors with requests (@lwis)
- Media Player: Google Play Music Desktop Player now supported (@GreenTurtwig)
- Notify: Twilio SMS platform added (@robbiet480)
- Device Tracker: Asus WRT will now default to SSH (but telnet is still an option) (@froz)
- Support Nest Protect smoke alarms (@khabi)
- Configurator: allow supplying a link to the user (@mnestor)
- Media Player: Roku now supported (@bah2830)
- Sensor: Last.FM now supported (@darookee, @GreenTurtwig)
- Notify: Amazon Lambda, SNS, SQS now supported (@robbiet480)
- Light: allow human readable colors in turn_on command (@robbiet480)
- YAML: new include dir options `!include_dir_merge_list` and `!include_dir_merge_named` (@happyleavesaoc)
- Media Player: LG Netcast TVs now supported (@wokar)
- Media Player: Allow enqueuing media on Sonos (@shaftoe)
- Notify: Ecobee thermostats now supported (@nkgilley)
- Sensor: mold indicator added (@xifle)
- Export to Logentries support added (@omgapuppy)
- Group more info dialog allows control of group domain if available (@fignuts)

Breaking changes:
- Asus WRT will now default to SSH with Telnet being an option

``` yaml
device_tracker:
  platform: asuswrt
  protocol: telnet
```

justyns committed:
Add 'purge_days' option to the history/recorder component

Issue https://github.com/balloob/home-assistant/issues/1337

When purge_days is set under the history component, recorder.py will
delete all events and states that are older than purge_days days ago.

Currently, this is only done once at start up.   A vacuum command is
also run to free up the disk space sqlite would still use after deleting
records.

justyns committed:
Add CONFIG_SCHEMA to verify config.  Move purge_days key name to
CONF_PURGE_DAYS so it can be changed easier later.

Use 'recorder' domain instead of 'history' domain.

Pass purge_days config directly into Recorder object instead of passing
the config object around.

justyns committed:
Add tests for purging old states and events

justyns committed:
Change sqlite queries to work with older versions of sqlite

balloob committed:
Version bump to 0.20.0.dev0

balloob committed:
Improve config validation error message

balloob committed:
Fix automation deprecation warning

turbokongen committed:
Fix for not recognizing Z-Wave thermostats (#2006)

* Fix for not recognizing thermostats

* Properly ignore zxt-120

* fix

balloob committed:
More fault tolerant discovery

balloob committed:
Add HVAC to demo

balloob committed:
Make HVAC naming consistent (#2017)

jazzaj committed:
Corrected link to documentation (#2022)

robbiet480 committed:
en_UK->en_GB. Closes #2019.

philipbl committed:
Fix problem with default channel

robbiet480 committed:
Minor Fitbit tweaks. Correct the copy, dont require auth on the routes, get the client_id/client_secret from fitbit.conf instead of the YAML

robbiet480 committed:
Fitbit flake8 and pylint fixes. Forgot to do it before pushing :(

robbiet480 committed:
Merge pull request #2023 from philipbl/fix_slack

Fix problem with Slack default channel

fabaff committed:
Upgrade flake8 to 2.5.4 (#2018)

kellerza committed:
Add a load_platform mechanism (#2012)

* discovery.load_platform method

* rm grep

nkgilley committed:
Bring back custom scan intervals and service for speedtest.net component (#1980)

* Bring back the functionality that was removed in PR 1717. This includes the speedtest service and the ability to define the scan times in the configuration file.  Restore default functionality of 1 scan per hour on the hour.

* remove unnecessary code.

robbiet480 committed:
Add a Jinja filter for relative time

Landrash committed:
Fixed minor miss-spelling (#2028)

Changed millileters to milliliters.
Changed case of mmol/l to mmol/L.

robbiet480 committed:
Remove humanize and use a relative time thing that @balloob found on Github

balloob committed:
Update README.rst

nkgilley committed:
Use Throttle on speedtest update (#2036)

* use throttle

* fix flake8

molobrakos committed:
log received mqtt messages (#2031)

balloob committed:
Make AND and OR conditions valid (#2037)

nkgilley committed:
Fix speedtest by removing Throttle and adding second parameter for track_time_change (#2040)

nkgilley committed:
fix away mode. issue 2032 (#2044)

happyleavesaoc committed:
Media Player - MPD: handle more exceptions (#2045)

kellerza committed:
Discovery listener on all EntityComponents (#2042)

nkgilley committed:
Add Ecobee notify platform (#2021)

* add send_message to ecobee via service call

* farcy fixes

* fix pydocstyle

* ecobee notify component

robbiet480 committed:
Add get_age tests

robbiet480 committed:
simplify the relative_time function

robbiet480 committed:
Stupid blank lines

robbiet480 committed:
Merge pull request #2027 from robbiet480/relative-time-filter

Add a Jinja filter for relative time

balloob committed:
Ignore RPI-RF in requirements_all

balloob committed:
Update rpi_rf.py

infamy committed:
update to latest base

infamy committed:
fixes for pep and delay start

kellerza committed:
Added QwikSwitch component & platforms (#1970)

* Added QwikSwitch platform

farcy - worst than my english teacher

* Clean up comments

* Import only inside functions

* Moved imports, no global var, load_platform

* add_device reworked

* Only serializable content on bus

* Fixed imports & removed some logging

happyleavesaoc committed:
supervisord sensor (#2056)

robbiet480 committed:
Google travel time improvements (#2047)

* Update google_travel_time.py

* Update google_travel_time.py

* pylint: disable=too-many-instance-attributes

* Add the mode to the title of the sensor

* Expose the travel mode on the sensor attributes

* Big improvements to the Google Travel Time sensor. Allow passing any options that Google supports in the options dict of your configuration. Deprecate travel_mode. Change name format to show the mode

* fu farcy

* Dynamically convert departure and arrival times

* Add a warning if user provides both departure and arrival times

* Add deprecation warning for travel_mode outside options and other minor fixes

* Use a copy of options dict to not overwrite the departure/arrival times constantly.

* Remove default travel_mode, but set default options.mode to driving

* Google doesnt let us query time in the past, so if the date we generate from a time string is in the past, add 1 day

* spacing fix

* Add config validation for all possible parameters

* flake8 and pylint fixes

robbiet480 committed:
Dont default to driving anymore, re: #2047

lwis committed:
Update Dockerfile to use OpenSSL 1.0.2h to resolve certificate issues (#2057)

infamy committed:
Fixes for farcy

infamy committed:
Fixes for farcy

mnestor committed:
YAML: add !include_named_dir and ! include_list_dir (#2054)

* add include_dir constructor for yaml parsing

* changed to allow for flat and name based directory including

* fixed ci errors

* changed flat to list

ishults committed:
Fix typo in system monitor ('recieved') (#2062)

Commit:
Add test for yaml enviroment

Danielhiversen committed:
Merge pull request #2068 from home-assistant/yaml_env

Add test for yaml enviroment

Danielhiversen committed:
Refactor get_age in util/dt (#2067)

Danielhiversen committed:
fix bugs in google travel time (#2069)

happyleavesaoc committed:
numeric state: validate multiple entities (#2066)

* validate multiple entities

* point to current entity

GreenTurtwig committed:
Google Play Music Desktop Player component (#1788)

* Added GPM Desktop Plaeyr component

* Updated requirements_all.txt

* Pylint fix

* Updated GPMDP.py to include @balloob's comments

* Updated to work with the latest version of GPMDP

* Removed setting "self._ws.recv()" as a variable

* Made line 52 shorter

* Updated to check weather it is connected or not

* Pylint and @balloob fixes

* Updated with simplified code and pylint fix

* Made `json.loads` shorter

* Pylint fix

robbiet480 committed:
Fix Google Voice documentation URL

robbiet480 committed:
Add lines for associated documentation PR

robbiet480 committed:
Add notify.twilio_sms component (#2070)

kellerza committed:
Resolved UI flicker, new config vars, brightness up to 255, fixed buttons, fixed race condition (#2072)

infamy committed:
Merge pull request #2059 from infamy/justyns-purge_old_data

Justyns purge old data

froz committed:
Device Tracker - ASUSWRT: Replaced telnet with ssh (#2079)

Khabi committed:
Support for Nest Protect smoke alarms (#2076)

* Support for Nest Protect smoke alarms

* Fixing formatting issues from tox

mnestor committed:
add link ability to configurator (#2035)

Commit:
Added support for the roku media player (#2046)

GreenTurtwig committed:
Last.fm sensor (#2071)

* Last.fm component

* Pylint fixes

* Last.fm component

* Pylint fixes

* Updated with `.coveragerc` and `requirements_all.txt`

* Pylint fixes

* Updated

* Pylint fix

* Pylint fix

robbiet480 committed:
Add AWS notify platforms (Lambda, SNS, SQS) (#2073)

* AWS SNS notify platform

* Attach kwargs as MessageAttributes

* Initial pass of AWS SQS platform

* Add Lambda notify platform

* Remove unused import

* Change single quotes to double quotes because I am crazy

* Forgot to run pydocstyle

* Improve context support for Lambda

* compress the message_attributes logic

Commit:
Round minutes to integer in google travel time, Fix issue #2080

Danielhiversen committed:
Merge pull request #2086 from home-assistant/time_travel_fix

Round minutes to integer in google travel time, Fix issue #2080

balloob committed:
Update frontend repo

Commit:
Better handle exceptions from Sonos players (#2085)

Sonos players can be dynamically set in various modes, for example
as TV players or Line-IN or straming from radios channels, therefore
some methods could not be available, and when invoked they cause
long exceptions to be logged. This partially solves the problem
reducing the output and logging some more informative error message

robbiet480 committed:
Accept human readable color names to change light colors (#2075)

* Add support for providing color_name which accepts a CSS3 valid, human readable string such as red or blue

* Forgot the schema validation!

* ugh farcy

* use html5_parse_legacy_color for more input options

* Add webcolors==1.5 to setup.py

* Block pylint no-member errors on tuple

* add color_name_to_rgb test

* whoops

* revert changes to individual platforms

* If color_name is set, pop it off params and set rgb_color with it

* Forgot to reset wink.py

* Import the legacy function as color_name_to_rgb directly

* reset test_color.py

* Improve light services.yaml

happyleavesaoc committed:
add some include_dir options (#2074)

* add some include_dir options

* validate, and extend instead of add

* add yaml include tests

froz committed:
Restored telnet as an option. Activate with config option 'protocol: telnet'. Default is ssh (#2096)

robbiet480 committed:
Catch a gntp networkerror (#2099)

robbiet480 committed:
Fix profile usage with aws notify platforms (#2100)

fabaff committed:
Upgrade psutil to 4.2.0 (#2101)

fabaff committed:
Upgrade python-telegram-bot to 4.1.1 (#2102)

Danielhiversen committed:
Fix bug in flaky rfxtrx test (#2107)

pavoni committed:
Handle region enter/leave with spaces.

pavoni committed:
Ignore acc: 0 updates.

balloob committed:
Merge pull request #2108 from home-assistant/owntracks_fixes

Owntracks fixes

pavoni committed:
Bump loop energy library version.

pavoni committed:
Merge pull request #2110 from home-assistant/bump_loop_energy

Bump loop energy library version.

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

Conflicts:
	homeassistant/const.py

persandstrom committed:
redirect daemon file descriptors (#2103)

jaharkes committed:
Allow for restart without using parent/child processes. (#1793)

* Allow for restart without using parent/child processes.

Assuming that we normally correctly shut down running threads and
release resources, we just do some minimal scrubbing of open file
descriptors and child processes which would stay around across an
exec() boundary.

* Use sys.executable instead of multiprocessing.spawn.get_executable()

* Limit how many file descriptors we try to close.

Don't even try to close on OSX/Darwin until we figure out how to
recognize guarded fds because the kernel will yell at us, and kill
the process.

* Use the close on exec flag on MacOS to clean up.

* Introduce a small process runner to handle restart on windows.

* Handle missing signal.SIGHUP on Windows.

wokar committed:
Added the lg_netcast platform to control a LG Smart TV running NetCast 3.0 or 4.0 (#2081)

* Added the `lgtv` platform to control a LG Smart TV running NetCast 3.0
(LG Smart TV models released in 2012) and NetCast 4.0 (LG Smart TV models released in 2013).

* Fixed multi-line docstring closing quotes

* Rename lgtv to lg_netcast

* Rename lgtv to lg_netcast

* Extracted class to control the LG TV into a separate Python package 'pylgnetcast' and changed requirements accordingly.

* regenerated requirements_all.txt with script

* now uses pylgnetcast v0.2.0 which uses the requests package for the communication with the TV

* fixed lint error: Catching too general exception Exception

turbokongen committed:
Fix fitbit KeyError (#2077)

* Fix fitbit KeyError

* Set units compared to temperature_unit

* Pass true or false for is_metric

Commit:
Add add_uri_to_queue support to (sonos) media player (#1946)

Sonos (SoCo) supports add_uri_to_queue capability, making it possible
to stream media available via HTTP for example. This patch extends
media_player component and sonos platform to support this feature

jaharkes committed:
Not sure why, but this fixed a bad filedescriptor error. (#2116)

nkgilley commented:
A couple of these aren't working for me.

- home-assistant/home-assistant#1793 Docker won't start
- home-assistant/home-assistant#2057 HA in Docker still has SSL errors with google travel time sensor.

Also added in this release is a new notify platform: ecobee. (home-assistant/home-assistant#2021)

Commit:
Add media_player.sonos_group_players service (#2087)

Sonos platform supports a `party mode` feature that groups all
available players into a single group, of which the calling player
will be the coordinator.

jaharkes committed:
Single process restart fixes (#2118)

* Ignore permission errors on setpgid.

When launched in a docker container we got a permission denied error
from setpgid.

* Don't fail if we find our own pidfile.

When we restart using exec we are running a new instance of home-assistant with
the same process id so we shouldn't be surprised to find an existing pidfile in
that case.

* Allow restart to work when started as python -m homeassistant.

When we are started with `python -m homeassistant`, the restart command line
becomes `python /path/to/hass/homeassistant/__main__.py`. But in that case the
python path includes `/path/to/hass/homeassistant` instead of `/path/to/hass`
and we fail on the first import.

Fix this by recognizing `/__main__.py` as part of the first argument and
injecting the proper path as PYTHONPATH environment before we start the new
home-assistant instance.

fabaff committed:
Docs (#2124)

* Add link to docs

* Update link

Commit:
Moldindicator Sensor (#1575)

* Adds MoldIndicator sensor platform

This sensor may be used to get an indication for possible mold growth in rooms.
It calculates the humidity at a pre-calibrated indoor point (wall, window).

* Automatic conversion to Fahrenheit for mold_indicator

* Minor change to critical temp label

* Fixed docstrings and styles

* Minor changes to MoldIndicator implementation

* Added first (non-working) implementation for mold_indicator test

* Small style changes

* Minor improvements to mold_indicator

* Completed unit test for mold indicator

* Fix to moldindicator initialization

* Adds missing period. Now that really matters..

* Adds test for sensor_changed function

robbiet480 committed:
Fix docstring issues with MoldIndicator

especially-relative committed:
Onkyo updates (#2084)

* use sane defaults for openzwave config

Use sane default if libopenzwave is installed. In most cases this will
mean that the zwave config path will not need to e manually specified.

* Resuming work on onkyo component

* Source control added to UI for onkyo receiver

Source will now display in the UI. Source mappings can be defined in the
config, and a rudimentary mapping is defined by default as a fallback.
When the onkyo source is updated, it will resolve to a defined name if
possible. This may break existing automations.

* fix lint errors

* Updated Onkyo receiver

Now takes an optional ip/name in additional to atempting to discover
deivces.

Source select will now take a sources mapping in the config. It will
provide default values if no source mapping is provided.

example:

- platform: onkyo
  host: 10.0.0.2
  name: receiver
  sources:
    HTPC: 'pc'
    Chromecast: 'aux1'
    Bluray: 'bd'
    Wii U: 'game'

* fix pylint error

* Use HA's error log instead of stack trace

* Flipped source mappings, code cleanup

kellerza committed:
Updated pyqwikswitch & QS<->HA UI behaviour (#2123)

* Updated pyqwikswitch & constants

* Disable too-many-locals

nkgilley committed:
Change color_RGB_to_xy formula & return brightness (#2095)

* Use RGB to XY calculations from Philips Hue developer site

* uppercase X,Y,Z

* rename cx,cy to x,y

* return brightness in color_RGB_to_xy

* remove try/catch

* update existing platforms using color_RGB_to_xy

* improve wemo w/ jaharkes suggestion

* allow brightness override of rgb_to_xy

ishults committed:
#2120 Fix hvac z-wave fan list (#2121)

* #2120 Fix hvac z-wave fan list

* Properly name methods

robbiet480 committed:
GTFS fixes (#2119)

* Change to official PyGTFS source

* Threading fixes for GTFS

* Actually pygtfs 0.1.3

* Update requirements_all.txt

* Update gtfs version

omgapuppy committed:
Add support for Logentries (#1945)

* Add support for Logentries

Supports sending has events to Logentries web hook endpoint
see logentries.com for more

Inspired by the Splunk component

* bugfix

* fix summary

* fix test

* fix logentries url and tests

* update tests

* mock token

* Bug fixes

* typo

* typo

* fix string splitting

* remove redundant backslash

balloob committed:
Upgrade Nest to 2.9.2 (#2126)

jaharkes committed:
Let systemd handle home-assistant process restarts. (#2127)

balloob committed:
Update frontend
